### PR TITLE
Don't crash when trying to write an unwritable keyconf.

### LINKED ISSUE #1235: QtOSError when key config is unreadable
When doing `qutebrowser --basedir base`, then `chmod 000 base/config/keys.conf` and starting it again we get:

```
==== Debug log ====
[...]
11:04:39 ERROR    keyboard   keyconf:_read:309 Failed to read key bindings!
Traceback (most recent call last):
  File "/home/florian/proj/qutebrowser/git/qutebrowser/config/parsers/keyconf.py", line 286, in _read
    with open(self._configfile, 'r', encoding='utf-8') as f:
PermissionError: [Errno 13] Permission denied: 'base/config/keys.conf'
[...]
11:04:39 DEBUG    init       app:init:161 Init done!
11:04:39 DEBUG    save       savemanager:save:98 Save of key-config requested - dirty True, save_on_exit False, is_exit False, force False -> True
11:04:39 DEBUG    destroy    keyconf:save:148 Saving key config to base/config/keys.conf
11:04:39 ERROR    misc       crashsignal:exception_hook:211 Uncaught exception
Traceback (most recent call last):
  File "/home/florian/proj/qutebrowser/git/qutebrowser/misc/savemanager.py", line 100, in save
    self._save_handler()
  File "/home/florian/proj/qutebrowser/git/qutebrowser/config/parsers/keyconf.py", line 149, in save
    with qtutils.savefile_open(self._configfile, encoding='utf-8') as f:
  File "/usr/lib/python3.4/contextlib.py", line 59, in __enter__
    return next(self.gen)
  File "/home/florian/proj/qutebrowser/git/qutebrowser/utils/qtutils.py", line 204, in savefile_open
    raise QtOSError(f)
qutebrowser.utils.qtutils.QtOSError: Existing file base/config/keys.conf is not writable
```

When `qutebrowser.conf` is unreadable we get an error message as expected.

### PR DESCRIPTION
**Pull request by Carpetsmoker**
Also change the logic in _load_default a wee bit so that it won't try to
write the keys.conf on startup.

Fixes #1235

### CODE REVIEW

File: qutebrowser/config/parsers/keyconf.py [hunk 1]
```diff
@@ -142,9 +142,14 @@ def _str_section_desc(self, sectname):
...
-        with qtutils.savefile_open(self._configfile, encoding='utf-8') as f:
-            data = str(self)
-            f.write(data)
+
+        try:
+            with qtutils.savefile_open(self._configfile,
+                                       encoding='utf-8') as f:
+                data = str(self)
+                f.write(data)
+        except (OSError, qtutils.QtOSError) as e:
```
Comment (The-Compiler) [inline qutebrowser/config/parsers/keyconf.py]: `QtOSError` inherits `OSError`, so I don't think it's needed to list them both here.

File: qutebrowser/misc/savemanager.py [hunk 2]
```diff
@@ -202,6 +202,6 @@ def save_command(self, *what):
             else:
                 try:
                     self.save(key, explicit=explicit, force=True)
-                except OSError as e:
+                except (OSError, utils.qtutils.QtOSError) as e:
```
Comment (The-Compiler) [inline qutebrowser/misc/savemanager.py]: See above

File: qutebrowser/config/parsers/keyconf.py (hunk 1 above)
Comment (Carpetsmoker) [inline qutebrowser/config/parsers/keyconf.py]: Ah, didn't realize that. I I'll go fix it.

**Comment by The-Compiler** [comment]
Thanks!

**Pull request merged**